data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/Router_ThreeRoutesIT.java

```diff
@@ -0,0 +1,168 @@
+    private static final String BETA_VALUE = "b";
+    private DataPrepperTestRunner dataPrepperTestRunner;
+    private InMemorySourceAccessor inMemorySourceAccessor;
+    private InMemorySinkAccessor inMemorySinkAccessor;
+
+    @BeforeEach
+    void setUp() {
+        dataPrepperTestRunner = DataPrepperTestRunner.builder()
+                .withPipelinesDirectoryOrFile("route/three-route.yaml")
+                .build();
+
+        dataPrepperTestRunner.start();
+        inMemorySourceAccessor = dataPrepperTestRunner.getInMemorySourceAccessor();
+        inMemorySinkAccessor = dataPrepperTestRunner.getInMemorySinkAccessor();
+    }
+
+    @AfterEach
+    void tearDown() {
+        dataPrepperTestRunner.stop();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            ALL_SOURCE_KEY,
+            ALPHA_BETA_SOURCE_KEY,
+            ALPHA_BETA_GAMMA_SOURCE_KEY
+    })
+    void sending_alpha_and_beta_events_sends_to_sinks_that_take_both(final String sourceKeyToReceiveAll) {
+        final List<Record<Event>> alphaEvents = createEvents(ALPHA_VALUE, 10);
+        final List<Record<Event>> betaEvents = createEvents(BETA_VALUE, 20);
+
+        final List<Record<Event>> allEvents = new ArrayList<>(alphaEvents);
+        allEvents.addAll(betaEvents);
+        Collections.shuffle(allEvents);
+
+        inMemorySourceAccessor.submit(TESTING_KEY, allEvents);
+
+        await().atMost(400, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+                    assertThat(inMemorySinkAccessor.get(sourceKeyToReceiveAll), not(empty()));
+                });
+
+        final List<Record<Event>> actualAllRecords = inMemorySinkAccessor.get(sourceKeyToReceiveAll);
+
+        assertThat(actualAllRecords.size(), equalTo(allEvents.size()));
+        assertThat(actualAllRecords, containsInAnyOrder(allEvents.toArray()));
+    }
+
+    @Test
+    void sending_alpha_and_beta_events_sends_to_the_sink_with_alpha_only_routes() {
+        final List<Record<Event>> alphaEvents = createEvents(ALPHA_VALUE, 10);
+        final List<Record<Event>> betaEvents = createEvents(BETA_VALUE, 20);
+
+        final List<Record<Event>> allEvents = new ArrayList<>(alphaEvents);
+        allEvents.addAll(betaEvents);
+        Collections.shuffle(allEvents);
+
+        inMemorySourceAccessor.submit(TESTING_KEY, allEvents);
+
+        await().atMost(400, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+                    assertThat(inMemorySinkAccessor.get(ALPHA_SOURCE_KEY), not(empty()));
+                });
+
+        final List<Record<Event>> actualAllRecords = inMemorySinkAccessor.get(ALPHA_SOURCE_KEY);
+
+        assertThat(actualAllRecords.size(), equalTo(alphaEvents.size()));
+
+        assertThat(actualAllRecords, containsInAnyOrder(allEvents.stream()
+                .filter(alphaEvents::contains).toArray()));
+    }
+
+    @Test
+    void sending_alpha_and_beta_events_sends_to_the_sink_with_beta_only_routes() {
```

Comment:
This seems redundant with the previous test case to me

Reply:
I had wondered about adding this as well. But, it does verify that a second route works in the `routes` list.

```
  - in_memory:
         testing_key: ConditionalRoutingIT_alpha_beta
         routes:
           - alpha
           - beta
```

Reply:
Wait. I think according to your scenario it should be ALPHA_BETA_SOURCE_KEY in the test body?

Reply:
Yes, you are correct. These tests are both covering that scenario. I'll create new PR to remove this test and drop that sink entirely.

Reply:
@chenqi0805, see #2061 for a fix.